src/crypto/crypto.cpp

```diff
@@ -88,12 +88,21 @@ namespace crypto {
     memcpy(&res, tmp, 32);
   }
 
-  static inline void hash_to_scalar(const void *data, size_t length, ec_scalar &res) {
+  void hash_to_scalar(const void *data, size_t length, ec_scalar &res) {
     cn_fast_hash(data, length, reinterpret_cast<hash &>(res));
     sc_reduce32(&res);
   }
 
   /* 
+   * copy 256bit data of ec point/scalar
+   */
+  void crypto_ops::copy(const ec_point &src, ec_point &dst) {
+    std::copy(src.data, src.data + 32, dst.data);
+  }
+  void crypto_ops::copy(const ec_scalar &src, ec_scalar &dst) {
+    std::copy(src.data, src.data + 32, dst.data);
+  }
+  /* 
    * generate public and secret keys from a random 256-bit integer
```

Comment:
I did this because simply assigning ec_point from one variable to another wouldn't copy the char data[32], but now I realized it does. So this should be unnecessary, it can be simply written as `dst = src`. Correct?

I'm so used to using `std::vector` or `std::array` and didn't familiarize myself enough with raw arrays... (I'm surprised because raw arrays themselves cannot be copied directly.)